test/strings/basic.jl

```diff
@@ -623,7 +626,87 @@ end
 
         @test prevind(strs[2], -1) == 0
         @test prevind(strs[2], -1, 1) == 0
+
+        @test prevind(strs[3], -1) == -2
+        @test prevind(strs[3], -1, 1) == -2
+        @test prevind(strs[4], -1) == -2
+        @test prevind(strs[4], -1, 1) == -2
+
+        @test prevind(strs[5], -4) == 0
+        @test prevind(strs[5], -4, 1) == 0
+        @test prevind(strs[6], -4) == 0
+        @test prevind(strs[6], -4, 1) == 0
+
+        @test prevind(strs[7], -4) == -5
+        @test prevind(strs[7], -4, 1) == -5
+
+        @test prevind(strs[8], -4) == 0
+        @test prevind(strs[8], -4, 1) == 0
     end
+
+    @test prevind(SubString("1234567", 1, 3), 10) == 3
+    @test prevind(SubString("1234567", 1, 3), 10, 1) == 3
+    @test prevind(SubString("1234567", 1, 3), 10, 2) == 2
+    @test prevind(SubString("1234567", 1, 3), 3) == 2
+    @test prevind(SubString("1234567", 1, 3), 2) == 1
+    @test prevind(SubString("1234567", 1, 3), 1) == 0
+    @test prevind(SubString("1234567", 1, 3), 0) == -1
+    @test prevind(SubString("1234567", 1, 3), 3, 1) == 2
+    @test prevind(SubString("1234567", 1, 3), 2, 1) == 1
+    @test prevind(SubString("1234567", 1, 3), 1, 1) == 0
+    @test prevind(SubString("1234567", 1, 3), 0, 1) == -1
+    @test prevind(SubString("1234567", 1, 3), 3, 2) == 1
+    @test prevind(SubString("1234567", 1, 3), 2, 2) == 0
+    @test prevind(SubString("1234567", 1, 3), 1, 2) == -1
+    @test prevind(SubString("1234567", 1, 3), 0, 2) == -2
+
+    @test nextind(SubString("1234567", 3), -6) == 1
+    @test nextind(SubString("1234567", 3), -6, 1) == 1
+    @test nextind(SubString("1234567", 3), -6, 2) == 2
+    @test nextind(SubString("1234567", 3), 1) == 2
+    @test nextind(SubString("1234567", 3), 2) == 3
+    @test nextind(SubString("1234567", 3), 3) == 4
+    @test nextind(SubString("1234567", 3), 4) == 5
+    @test nextind(SubString("1234567", 3), 1, 1) == 2
+    @test nextind(SubString("1234567", 3), 2, 1) == 3
+    @test nextind(SubString("1234567", 3), 3, 1) == 4
+    @test nextind(SubString("1234567", 3), 4, 1) == 5
+    @test nextind(SubString("1234567", 3), 1, 2) == 3
+    @test nextind(SubString("1234567", 3), 2, 2) == 4
+    @test nextind(SubString("1234567", 3), 3, 2) == 5
+    @test nextind(SubString("1234567", 3), 4, 2) == 6
+
+    @test prevind(SubString(GenericString("1234567"), 1, 3), 10) == 3
```

Comment:
Can't these be merged with the previous tests using a loop?

Reply:
this is what I have done previously, but the discontinuity at `1` and `endof(s)` made it problematic (the test code would be very messy).

Reply:
I must be missing something. These lines are identical to those above except for the `GenericString` part, so what's the problem?

Reply:
I have put into the loop all similar cases and left the differences below.